prow/crier/reporters/pubsub/reporter.go

```diff
@@ -98,12 +98,16 @@ func (c *Client) Report(_ *logrus.Entry, pj *prowapi.ProwJob) ([]*prowapi.ProwJo
 	message := c.generateMessageFromPJ(pj)
 
 	ctx := context.Background()
+	// TODO: Consider caching the pubsub client.
```

Comment:
This definitely seems like an excellent idea for a subsequent PR. Current design is incredibly wasteful.